rir/src/interpreter/interp.cpp

```diff
@@ -491,6 +491,264 @@ RIR_INLINE SEXP legacyCall(const CallContext& call, Context* ctx) {
+                        } else {
+                            REAL(res)[pos++] = INTEGER(args[i])[j];
+                        }
+                    }
+                } else {
+                    INTEGER(res)[pos++] = INTEGER(args[i])[j];
+                }
+            }
+        }
+        return res;
+    }
+
+    case 109: { //"vector"
+        if (nargs != 2)
+            return nullptr;
+        if (TYPEOF(args[0]) != STRSXP)
+            return nullptr;
+        if (XLENGTH(args[0]) != 1)
+            return nullptr;
+        size_t length = -1;
+        if (IS_SIMPLE_SCALAR(args[1], INTSXP)) {
+            if (INTEGER(args[1])[0] == NA_INTEGER)
+                return nullptr;
+            length = INTEGER(args[1])[0];
+        } else if (IS_SIMPLE_SCALAR(args[1], REALSXP)) {
+            if (REAL(args[1])[0] == NA_REAL)
+                return nullptr;
+            length = REAL(args[1])[0];
+        } else {
+            return nullptr;
+        }
+        int type = str2type(CHAR(args[0]));
+
+        switch (type) {
+        case LGLSXP:
+        case INTSXP: {
+            auto res = allocVector(type, length);
+            Memzero(INTEGER(res), length);
+            return res;
+        }
+        case CPLXSXP: {
+            auto res = allocVector(type, length);
+            Memzero(COMPLEX(res), length);
+            return res;
+        }
+        case RAWSXP: {
+            auto res = allocVector(type, length);
+            Memzero(RAW(res), length);
+            return res;
+        }
+        case REALSXP: {
+            auto res = allocVector(type, length);
+            Memzero(REAL(res), length);
+            return res;
+        }
+        case STRSXP:
+        case EXPRSXP:
+        case VECSXP:
+            return allocVector(type, length);
+        case LISTSXP:
+            if (length > INT_MAX)
+                return nullptr;
+            return allocList((int)length);
+        default:
+            return nullptr;
+        }
+        assert(false);
+    }
+
+    case 412: { // "list"
+        auto res = Rf_allocVector(VECSXP, nargs);
+        for (size_t i = 0; i < nargs; ++i)
+            SET_VECTOR_ELT(res, i, args[i]);
+        return res;
+    }
+
+    case 399: { // "is.vector"
+        if (nargs != 1)
+            return nullptr;
+
+        auto arg = args[0];
+        if (XLENGTH(arg) != 1)
+            return nullptr;
+        return TYPEOF(arg) == VECSXP ? R_TrueValue : R_FalseValue;
+    }
+
+    case 395: { // "is.na"
+        if (nargs != 1)
+            return nullptr;
+
+        auto arg = args[0];
+        if (XLENGTH(arg) != 1)
+            return nullptr;
+
+        switch (TYPEOF(arg)) {
+        case INTSXP:
+            return INTEGER(arg)[0] == NA_INTEGER ? R_TrueValue : R_FalseValue;
+        case LGLSXP:
+            return LOGICAL(arg)[0] == NA_LOGICAL ? R_TrueValue : R_FalseValue;
+        case REALSXP:
+            return ISNAN(REAL(arg)[0]) ? R_TrueValue : R_FalseValue;
```

Comment:
(This might be a mistake I made), why not `REAL(arg)[0] == NA_REAL`?

Reply:
no, because there are many `NAN` values in the double type. Only one of them is the `NA_REAL`. I am not sure where the `NA_REAL` is used in particular. But at least this function returns true for all na's.